Packs/AggregatedScripts/Scripts/GetEndpointData/GetEndpointData.py

```diff
@@ -659,178 +876,173 @@ def get_raw_endpoints(output_keys: list[str], raw_context: list[dict[str, Any]])
-    return output_mapping
 
-
-def cylance_filtering(endpoints: list[dict[str, Any]], args: dict[str, Any]) -> list[dict[str, Any]]:
-    filtered_endpoints = []
-    hostnames = to_list(args.get("agent_hostname", []))
-    if not hostnames:
-        return endpoints
-    for endpoint in endpoints:
-        endpoint_hostname = endpoint["Hostname"]["Value"]
-        if endpoint_hostname in hostnames:
-            filtered_endpoints.append(endpoint)
-    return filtered_endpoints
+    for endpoint_id, endpoint_ip, endpoint_hostname in zipped_args:
+        if endpoint_id not in ids and endpoint_ip not in ips and endpoint_hostname not in hostnames:
+            endpoint_not_found = filter_empty_values({"ID": endpoint_id, "Hostname": endpoint_hostname, "IPAddress": endpoint_ip})
+            endpoints_not_found.append(endpoint_not_found)
+    return endpoints_not_found
 
 
-def active_directory_post(endpoints: list[dict[str, Any]], args: dict[str, Any]) -> list[dict[str, Any]]:
+def active_directory_post(
+    self: EndpointCommandRunner, endpoints: list[dict[str, Any]], args: dict[str, Any]
+) -> list[dict[str, Any]]:
     fixed_endpoints = []
     for endpoint in endpoints:
-        endpoint_hostname = endpoint["Hostname"]["Value"]
+        endpoint_hostname = endpoint["Hostname"]
         if isinstance(endpoint_hostname, str):
             fixed_endpoints.append(endpoint)
         elif isinstance(endpoint_hostname, list) and len(endpoint_hostname) == 1:
-            endpoint["Hostname"]["Value"] = endpoint_hostname[0]
+            endpoint["Hostname"] = endpoint_hostname[0]
             fixed_endpoints.append(endpoint)
         else:
             raise ValueError("Invalid hostname")
     return fixed_endpoints
 
 
+def generic_endpint_post(
+    self: EndpointCommandRunner, endpoints: list[dict[str, Any]], args: dict[str, Any]
+) -> list[dict[str, Any]]:
+    endpoints_to_return = []
+    for endpoint in endpoints:
+        brand = endpoint["Brand"]
+        if brand in Brands.get_all_values() and self.module_manager.is_brand_available(Command(brand, "", [], {}, {})):
+            # If the brand is in the brands, we dno't need if from the generic command
+            demisto.debug(f"Skipping generic endpoint with brand: '{brand}'")
+        else:
+            endpoints_to_return.append(endpoint)
+    return endpoints_to_return
+
+
 """ MAIN FUNCTION """
 
 
-def main():
+def main():  # pragma: no cover
     try:
         args = demisto.args()
-        agent_ids = argToList(args.get("agent_id", []))
-        agent_ips = argToList(args.get("agent_ip", []))
-        agent_hostnames = argToList(args.get("agent_hostname", []))
+        endpoint_ids = argToList(args.get("endpoint_id", []), transform=convert_none_to_empty_string)
+        endpoint_ips = argToList(args.get("endpoint_ip", []), transform=convert_none_to_empty_string)
+        endpoint_hostnames = argToList(args.get("endpoint_hostname", []), transform=convert_none_to_empty_string)
```

Comment:
```suggestion
        endpoint_ids = argToList(args.get("endpoint_id", [])) or ""
        endpoint_ips = argToList(args.get("endpoint_ip", []) or ""
        endpoint_hostnames = argToList(args.get("endpoint_hostname", []) or ""
```

and then we can remove the convert_none_to_empty_string function

Reply:
The reason is that only if it is `None` do we want to convert it to an empty string, but if it is, for example, an empty `Dict`, we want to leave it as it is.